Fix margin is recognized as unsupported property gridlayout

qtdzz opened the pull request:
* Add "margin", "margin-left", "margin-right","margin-top","margin-bottom" to GridLayout#getCustomAttributes otherwise these attributes will be considered as unsupported.
* Fix for the issue: https://github.com/vaadin/framework8-issues/issues/465

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/103)
<!-- Reviewable:end -->

Commit:
Fix margin in declarative format is recognized as unsupported property gridlayout

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=103) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=103) before we can accept your contribution.<br/>

ahie commented:
Please provide a test for this fix as well.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103#-:-KYXgtvKsiydNPYKYrD-:b-ikvrtj)*
<!-- Sent from Reviewable.io -->

pleku commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYXm7M4j4geeZgWvInd:b-9r5pw) (no related file):*
Could you please include these attributes in the test `GridLayoutDeclarativeTest` ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

qtdzz commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYN1u-Md60wmczAlXs:bi2mn7y) (no related file):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Could you please include these attributes in the test `GridLayoutDeclarativeTest` ?
</blockquote></details>

Hi. I update the PR with the test but it seems like I'm using the wrong code formatter. I couldn't find the codestyle formatter in README as instruction.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYRiQs_oGlntXxh-nf:b-j0v8pm) (no related file):*
<details><summary><i>Previously, qtdzz (Tien Nguyen) wrote…</i></summary><blockquote>

Hi. I update the PR with the test but it seems like I'm using the wrong code formatter. I couldn't find the codestyle formatter in README as instruction.
</blockquote></details>

Yeah should work out-of-the-box if you import the project to eclipse neon ... I can do a checkout and reformat it when I have time

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYYKPvFVjfsLZfs-Fu:blf5tco) (no related file):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Yeah should work out-of-the-box if you import the project to eclipse neon ... I can do a checkout and reformat it when I have time
</blockquote></details>

Thought I had a chance today, but  nope. Would appreciate if you could fix the formatting since it is quite cumbersome to checkout another fork and push to it ...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

qtdzz commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYg7jQH5d0-irBqZUB:b81iudu) (no related file):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Thought I had a chance today, but  nope. Would appreciate if you could fix the formatting since it is quite cumbersome to checkout another fork and push to it ...
</blockquote></details>

I updated the PR with new eclipse instance, new workspace and added test without formatting code. If I use reformat function of eclipse, it will mess the whole test class. Hope it will be fine now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

Legioth commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYl7fVyczmCWtQosUK:bgetkm9) (no related file):*
<details><summary><i>Previously, qtdzz (Tien Nguyen) wrote…</i></summary><blockquote>

I updated the PR with new eclipse instance, new workspace and added test without formatting code. If I use reformat function of eclipse, it will mess the whole test class. Hope it will be fine now
</blockquote></details>

What Pekka meant to say is that you need to import the formatting settings from `eclipse/VaadinJavaConventions.xml` and configure the project (or your entire workspace) to use those settings.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

qtdzz commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYmb9wF8owKUsJ7dte:b-ll6x52) (no related file):*
<details><summary><i>Previously, Legioth (Leif Åstrand) wrote…</i></summary><blockquote>

What Pekka meant to say is that you need to import the formatting settings from `eclipse/VaadinJavaConventions.xml` and configure the project (or your entire workspace) to use those settings.
</blockquote></details>

Thanks Leif. I reformatted it. Sorry for bothering you guys :(

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

Legioth commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/103#-KYXm7M4j4geeZgWvInc:-KYYodXdvbsnp51fmhK7:banggxm) (no related file):*
<details><summary><i>Previously, qtdzz (Tien Nguyen) wrote…</i></summary><blockquote>

Thanks Leif. I reformatted it. Sorry for bothering you guys :(
</blockquote></details>

Sorry for wasting your time by not spelling out those tings clearly in the readme :(

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->

pleku commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/103)*
<!-- Sent from Reviewable.io -->
